src/webgpu/util/device_pool.ts

```diff
@@ -60,10 +60,17 @@ export class DevicePool {
       // (Hopefully if the device was lost, it has been reported by the time endErrorScopes()
       // has finished (or timed out). If not, it could cause a finite number of extra test
       // failures following this one (but should recover eventually).)
-      const lostReason = holder.lostReason;
-      if (lostReason !== undefined) {
-        // Fail the current test.
-        unreachable(`Device was lost: ${lostReason}`);
+      const lost = holder.lost;
+      if (lost !== undefined) {
+        if (lost.reason !== 'destroyed') {
+          // Fail the current test.
+          unreachable(`Device was unexpectedly lost: ${lost.message}`);
+        }
+        if (holder === this.defaultHolder) {
+          this.defaultHolder = 'uninitialized';
+        } else {
+          this.nonDefaultHolders.deleteByDevice(holder.device);
+        }
       }
     } catch (ex) {
       // Any error that isn't explicitly TestFailedButDeviceReusable forces a new device to be
```

Comment:
Below on line 84 there's a MAINTENANCE_TODO to device.destroy() after a device is discarded, to clean up resources. I think we should add that now.

But guard it behind `if ('destroy' in device)` just so we don't break the entire CTS for implementations that haven't implemented device.destroy() yet.

Reply:
Done